ui/pages/confirmations/components/confirm/info/dapp-swap-comparison-utils.ts

```diff
@@ -0,0 +1,167 @@
+      },
+      {
+        name: 'inputs',
+        type: 'bytes[]',
+      },
+    ],
+    name: 'execute',
+    type: 'function',
+  },
+];
+
+const getWordsFromInput = (input: string) => {
+  return input?.slice(2).match(/.{1,64}/gu) ?? [];
+};
+
+const wordToAddress = (word?: string) => {
+  if (!word) {
+    return '';
+  }
+  return addHexPrefix(word?.slice(24));
+};
+
+const wordToAmount = (word?: string) => {
+  if (!word) {
+    return '';
+  }
+  const amount = word?.replace(/^0+/u, '') ?? '';
+  return addHexPrefix(amount);
+};
+
+const parseTransactionData = (data?: string) => {
+  const contractInterface = new Interface(ABI);
+
+  let parsedTransactionData: TransactionDescription;
+
+  try {
+    parsedTransactionData = contractInterface.parseTransaction({
+      data: data as Hex,
+    });
+  } catch (error) {
+    return { inputs: [], commandBytes: [] };
+  }
+
+  const { commands } = parsedTransactionData.args;
+  const { inputs } = parsedTransactionData.args;
+  const commandBytes = commands.slice(2).match(/.{1,2}/gu) ?? [];
+
+  return { inputs, commandBytes };
+};
+
+export const getDataFromSwap = (
+  chainId: Hex,
+  amount?: string,
+  data?: string,
+) => {
+  let quotesInput;
+  let amountMin;
+  const erc20TokenAddresses = [];
+  const { commandBytes, inputs } = parseTransactionData(data);
+
+  const sweepIndex = commandBytes.findIndex(
+    (commandByte: string) => commandByte === '04',
+  );
+
+  if (sweepIndex >= 0) {
+    const words = getWordsFromInput(inputs[sweepIndex]);
+    amountMin = wordToAmount(words[2]);
+    erc20TokenAddresses.push(wordToAddress(words[0]));
+    quotesInput = {
+      walletAddress: wordToAddress(words[1]),
+      srcChainId: chainId,
+      destChainId: chainId,
+      srcTokenAddress: getNativeTokenAddress(chainId),
+      destTokenAddress: wordToAddress(words[0]),
+      srcTokenAmount: amount ?? '0x0',
+      gasIncluded: false,
+      gasIncluded7702: false,
+    };
+  } else {
+    const seaportIndex = commandBytes.findIndex(
+      (commandByte: string) => commandByte === '10',
+    );
+
+    if (seaportIndex >= 0) {
+      const words = getWordsFromInput(inputs[seaportIndex]);
+      amountMin = wordToAmount(words[13]);
+      erc20TokenAddresses.push(wordToAddress(words[10]));
+      erc20TokenAddresses.push(wordToAddress(words[16]));
+      quotesInput = {
+        walletAddress: wordToAddress(words[28]),
+        srcChainId: chainId,
+        destChainId: chainId,
+        srcTokenAddress: wordToAddress(words[10]),
+        destTokenAddress: wordToAddress(words[16]),
+        srcTokenAmount: wordToAmount(words[12]),
+        gasIncluded: false,
+        gasIncluded7702: false,
+      };
+    }
+  }
+
+  return { quotesInput, amountMin, erc20TokenAddresses };
+};
+
+export const getBestQuote = (
+  quotes: QuoteResponse[],
+): QuoteResponse | undefined => {
+  let selectedQuoteIndex = -1;
+  let highestMinDestTokenAmount = '-1';
+
+  quotes.forEach((quote, index) => {
+    if (
```

Comment:
Unfortunately we can't assume it's the best just because the min is higher, as the simulated value could be lower.

I think we'd have to perform our math on all the quotes to identify the best based on total cost?

Reply:
What will be total cost calculation ?